packages/expo-barcode-scanner/src/BarCodeScanner.tsx

```diff
@@ -138,6 +144,15 @@ export class BarCodeScanner extends React.Component<BarCodeScannerProps> {
     barCodeTypes: Object.values(BarCodeType),
   };
 
+  componentDidMount(): void {
+    if (!warnedAboutDeprecation) {
+      console.warn(
+        'BarCodeScanner has been deprecated and will be removed in a future SDK version. Please use `expo-camera` instead'
```

Comment:
we should have an fyi page that explains how to migrate from barcodescanner to expo-camera, can you write one up and add a link to it in this warning @alanjhughes? to create an fyi page, just add a markdown file to https://github.com/expo/fyi

Reply:
No problem 👍